src/libraries/System.Runtime.Numerics/src/System/Numerics/Decimal64.cs

```diff
@@ -0,0 +1,1996 @@
+        // Binary:                0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000
+        // Split into sections:   0 | 000_0000_000 | 0_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000
+        // Section labels:        a | b          | c
+        //
+        // a. Sign bit.
+        // b. Biased Exponent, which is q + ExponentBias (398). 000_0000_000 == 0 == -398 + 398, so this is encoding a q of -398.
+        // c. Significand, set to 0.
+        //
+        // Encoded value:         0 x 10^-398
+
+        private const ulong PositiveZeroBits = 0x0000_0000_0000_0000;
+        private const ulong NegativeZeroBits = SignMask | PositiveZeroBits;
+
+
+        // Epsilon Bits
+        // Hex:                   0x0000_0000_0000_0001
+        // Binary:                0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0001
+        // Split into sections:   0 | 000_0000_000 | 0_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0001
+        // Section labels:        a | b          | c
+        //
+        // a. Sign bit.
+        // b. Biased Exponent, which is q + ExponentBias (398). 000_0000_000 == 0 == -398 + 398, so this is encoding a q of -398.
+        // c. Significand, set to 1.
+        //
+        // Encoded value:         1 x 10^-398
+
+        private const ulong EpsilonBits = 0x0000_0000_0000_0001;
+
+        // PositiveInfinityBits
+        // Hex:                   0x7800_0000_0000_0000
+        // Binary:                0111_1000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000
+        // Split into sections:   0 | 111_1000_0000_00 | 00_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000
+        // Section labels:        a | b            | c
+        //
+        // a. Sign bit.
+        // b. Combination field G0 through G12. G0-G4 == 11110 encodes infinity.
+        // c. Trailing significand.
+        // Note: Canonical infinity has everything after G5 set to 0.
+
+        private const ulong PositiveInfinityBits = 0x7800_0000_0000_0000;
+        private const ulong NegativeInfinityBits = SignMask | PositiveInfinityBits;
+
+        // QNanBits Bits
+        // Hex:                   0x7C00_0000_0000_0000
+        // Binary:                0111_1100_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000
+        // Split into sections:   0 | 111_1100_0000_00 | 00_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000
+        // Section labels:        a | b            | c
+        //
+        // a. Sign bit (ignored for NaN).
+        // b. Combination field G0 through G12. G0-G4 == 11111 encodes NaN.
+        // c. Trailing significand. Can be used to encode a payload, to distinguish different NaNs.
+        // Note: Canonical NaN has G6-G12 as 0 and the encoding of the payload also canonical.
+
+        private const ulong QNanBits = 0x7C00_0000_0000_0000; // TODO I used a "positive" NaN here, should it be negative?
```

Comment:
I'd probably keep it as positive for the "bit mask"

But we should probably return `-QNaN` for the "canonincal" version, so it matches what happens for the binary `QNaN`

Reply:
Sounds good, although I'm curious how we arrived at that decision. I think the intel library has positive `QNaN` as their "canonical" `NaN`, for example, this early return in bid64_add: 
![image](https://user-images.githubusercontent.com/18474647/218582969-d2e2e3fe-2121-4f4b-a68b-c0862e63b729.png)

Is it reasonable to return `-QNaN` for cases like this?

Reply:
Because x86 hardware returns `-QNaN` for binary floats, so it's been the historical "norm" that people see